isobus/include/isobus/isobus/isobus_preferred_addresses.hpp

```diff
@@ -0,0 +1,210 @@
+//================================================================================================
+/// @file isobus_preferred_addresses.hpp
+///
+/// @brief This is a reference for control function's preferred addresses as defined by
+/// ISO 11783-11 and/or SAE. Preferred addresses are industry group specific.
+/// You should use these when your are creating a control function that is a well known
+/// function type, but if your control function doesn't arbitrate for that address, the
+/// stack will claim for you in the dynamic address range.
+///
+/// @author Adrian Del Grosso
+///
+/// @copyright 2024 The Open-Agriculture Developers
+//================================================================================================
+#ifndef ISOBUS_PREFERRED_ADDRESSES_HPP
+#define ISOBUS_PREFERRED_ADDRESSES_HPP
+
+#include <cstdint>
+
+namespace isobus
+{
+	/// @brief This namespace contains all defined preferred addresses for control functions, which
+	/// should be used when creating a control function that is a well known function type.
+	namespace preferred_addresses
+	{
+		/// @brief Industry Group 1 applies to on-highway equipment. This
+		/// This enumerates all preferred addresses for this industry group.
+		enum class IndustryGroup1 : std::uint8_t
```

Comment:
thought: since I'd imagine this only to be used as a lookup table, could it be more user friendly to implicitly convert to `std::uint8_t` here? So we can just use 

```c++
	auto TestInternalECU = isobus::InternalControlFunction::create(TestDeviceNAME, isobus::preferred_address::IndustryGroup2::TaskController_MappingComputer, 0);
```

instead of 

```c++
	auto TestInternalECU = isobus::InternalControlFunction::create(TestDeviceNAME, static_cast<std::uint8_t>(isobus::preferred_address::IndustryGroup2::TaskController_MappingComputer), 0);
```

I know Sonar won't like it, but this might be an exception?

Reply:
Yep, I agree that does seem nicer (sorry sonar). I will make them regular enums instead.

Reply:
To get this behavior to work without redefinition errors (there are multiple enumerators with the same name) I had to do:

```
namespace preferred_addresses
{
	namespace IndustryGroup1
	{
		enum PreferredAddress : std::uint8_t
		{
```
For example